ambuild2/frontend/v2_2/cpp/builders.py

```diff
@@ -0,0 +1,602 @@
+            self.argv = ['ambuild_objcopy_wrapper.sh', self.outputFile] + self.argv
+
+    def link(self, context, folder, inputs):
+        # The existence of .ilk files on Windows does not seem reliable, so we
+        # treat it as "shared" which does not participate in the DAG (yet).
+        shared_outputs = []
+        if self.linker_.behavior == 'msvc':
+            if not isinstance(self, StaticLibrary) and '/INCREMENTAL:NO' not in self.argv:
+                shared_outputs += [self.name_ + '.ilk']
+
+        ignore, outputs = context.AddCommand(inputs = inputs,
+                                             argv = self.argv,
+                                             outputs = self.linker_outputs,
+                                             folder = folder,
+                                             weak_inputs = self.compiler.weaklinkdeps,
+                                             shared_outputs = shared_outputs,
+                                             env_data = self.compiler.env_data)
+        if not self.debug_entry and self.compiler.symbol_files:
+            if self.linker_.behavior != 'msvc' and self.compiler.symbol_files == 'bundled':
+                self.debug_entry = outputs[0]
+            else:
+                self.debug_entry = outputs[-1]
+        return outputs[0], self.debug_entry
+
+class Program(BinaryBuilder):
+    def __init__(self, compiler, name):
+        super(Program, self).__init__(compiler, name)
+
+    @staticmethod
+    def buildName(compiler, name):
+        return compiler.vendor.nameForExecutable(name)
+
+    @property
+    def type(self):
+        return 'program'
+
+    def generateBinary(self, cx, files):
+        return self.compiler.vendor.programLinkArgv(
+            cmd_argv = self.linker_argv_,
+            files = files,
+            linkFlags = self.linkFlags(cx),
+            symbolFile = self.name_ if self.compiler.symbol_files else None,
+            outputFile = self.outputFile)
+
+class Library(BinaryBuilder):
+    def __init__(self, compiler, name):
+        super(Library, self).__init__(compiler, name)
+
+    @staticmethod
+    def buildName(compiler, name):
+        return compiler.vendor.nameForSharedLibrary(name)
+
+    @property
+    def type(self):
+        return 'library'
+
+    def generateBinary(self, cx, files):
+        return self.compiler.vendor.libLinkArgv(
+            cmd_argv = self.linker_argv_,
+            files = files,
+            linkFlags = self.linkFlags(cx),
+            symbolFile = self.name_ if self.compiler.symbol_files else None,
+            outputFile = self.outputFile)
+
+class StaticLibrary(BinaryBuilder):
+    def __init__(self, compiler, name):
+        super(StaticLibrary, self).__init__(compiler, name)
+
+    @staticmethod
+    def buildName(compiler, name):
+        return compiler.vendor.nameForStaticLibrary(name)
+
+    @property
+    def type(self):
+        return 'static'
```

Comment:
Replace this with an enum.
For ```setup.py```, inject the following:
```
insall_requires = ['enum34;python_version<"3.4"']
```
The PyPi page for it SAYS it supports Python 2.4-3.3